rust/agama-lib/src/error.rs

```diff
@@ -16,6 +16,10 @@ pub enum ServiceError {
     Anyhow(#[from] anyhow::Error),
     #[error("Wrong user parameters: '{0:?}'")]
     WrongUser(Vec<String>),
+    #[error("Registration failed: '{0}'")]
+    FailedRegistration(String),
+    #[error("Failed to find these patterns: '{0:?}'")]
```

Comment:
```suggestion
    #[error("Failed to find these patterns: {0:?}")]
```

The Debug format will use `["crocheting"]` already, no need for more quotes